src/browser-connection/index.js

```diff
@@ -34,12 +35,16 @@ export default class BrowserConnection extends EventEmitter {
         this.idleUrl      = `${gateway.domain}/browser/idle/${this.id}`;
         this.statusUrl    = `${gateway.domain}/browser/status/${this.id}`;
 
+        this.on('error', () => {
+            this.disconnected = true;
```

Comment:
We should set this flag in `close` function. Because currently I can manually close connection and pass it to the runner and it will not cause early error.